ladybug/legend.py

```diff
@@ -0,0 +1,877 @@
+                         'segment_width', 'text_height', 'font')
+        default_keys = ('is_number_of_segments_default', 'is_title_default',
+                        'is_base_plane_default', 'is_segment_height_default',
+                        'is_segment_width_default', 'is_text_height_default')
+        for key in optional_keys:
+            if key not in data:
+                data[key] = None
+        for key in default_keys:
+            if key not in data:
+                data[key] = False
+
+        colors = None
+        if data['colors'] is not None:
+            colors = [Color.from_json(col) for col in data['colors']]
+        base_plane = None
+        if data['base_plane'] is not None:
+            base_plane = Plane.from_dict(data['base_plane'])
+
+        leg_par = cls(data['min'], data['max'], data['number_of_segments'],
+                      colors, data['continuous_colors'], data['continuous_legend'],
+                      data['title'], data['ordinal_dictionary'],
+                      data['number_decimal_places'], data['include_larger_smaller'],
+                      data['vertical_or_horizontal'], base_plane,
+                      data['segment_height'], data['segment_width'],
+                      data['text_height'], data['font'])
+        leg_par._is_number_of_segments_default = data['is_number_of_segments_default']
+        leg_par._is_title_default = data['is_title_default']
+        leg_par._is_base_plane_default = data['is_base_plane_default']
+        leg_par._is_segment_height_default = data['is_segment_height_default']
+        leg_par._is_segment_width_default = data['is_segment_width_default']
+        leg_par._is_text_height_default = data['is_text_height_default']
+        return leg_par
+
+    @property
+    def min(self):
+        """Get or set legend minimum."""
+        return self._min
+
+    @min.setter
+    def min(self, minimum):
+        if minimum is not None:
+            assert isinstance(minimum, (float, int)), \
+                'Expected number for min. Got {}.'.format(type(minimum))
+            if self._max is not None:
+                assert minimum <= self._max, \
+                    'Input min is greater than input max. {} > {}.'.format(
+                        minimum, self._max)
+        self._min = minimum
+
+    @property
+    def max(self):
+        """Get or set legend maximum."""
+        return self._max
+
+    @max.setter
+    def max(self, maximum):
+        if maximum is not None:
+            assert isinstance(maximum, (float, int)), \
+                'Expected number for max. Got {}.'.format(type(maximum))
+            if self._min is not None:
+                assert maximum >= self._min, \
+                    'Input max is less than input min. {} < {}.'.format(
+                        maximum, self._min)
+        self._max = maximum
+
+    @property
+    def number_of_segments(self):
```

Comment:
I originally used `number_of_x` for a lot of cases but came to realize `x_count` is shorter and started using that. We should probably use one of them to keep the code consistent.

Reply:
Took me some time to realize what you meant by `x` but now I see that you were just saying that it's a general rule you use across all property names with that structure.  Ok, I will change it to `segment_count`

Reply:
I'll do the same thing with `number_decimal_places`. `decimal_count` is much better.